ploomber/jupysql
# Feat: Telemetry - Collect Connection info (dialect, driver, server version)

## Context

We want to have a overview of what database & version that our users are using

## Describe your changes

1. Collect database connection info (dialect, driver, server version) in below command execution into telegram:
- `execute`
- `boxplot`
- `histogram`
- `data-frame`

2. Add `get_curr_connection_info` in sql.Connection class

3. Add Unit Test - `test_get_curr_connection_info`

4. Add Integration Test for 
- postgreSQL
- mySQL
- mariaDB
- SQLite
- duckDB

Example:
1. Execute `%sql postgresql+psycopg2://...`
2. Will see below `dialect_meta` in posthug

<img width="1538" alt="Screenshot 2023-02-21 at 1 27 54 PM" src="https://user-images.githubusercontent.com/123580782/220429090-c21c91c4-d288-4a0f-a3ff-c3907c2b4a50.png">


```
"connection_info": {
    "dialect": "postgresql",
    "driver": "psycopg2",
    "server_version_info": [
        15,
        1
    ]
}
```


## Issue ticket number and link
Closes #113

## Checklist before requesting a review

- [x] I have added thorough [tests](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#testing) (when necessary).
- [x] I have added the right documentation in the [docstring](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#documenting-changes-and-new-features) and [changelog](https://github.com/ploomber/contributing/blob/main/CONTRIBUTING.md#changelog) (when needed)




----
:books: Documentation preview :books:: https://jupysql--126.org.readthedocs.build/en/126/
